lib/common/memory/src/madvise.rs

```diff
@@ -117,4 +142,22 @@ impl Madviseable for memmap2::MmapMut {
         log::debug!("Ignore {advice:?} on this platform");
         Ok(())
     }
+
+    fn populate(&self) -> io::Result<()> {
+        #[cfg(target_os = "linux")]
+        self.advise(memmap2::Advice::PopulateRead)?;
```

Comment:
This option requires Linux 5.14+. Some long-term supported distros (e.g. RHEL 8 and Ubuntu 20.04) still have older kernels. Maybe it would make sense to check beforehand if it is supported and fallback to the loop implementation if it isn't. Quote from [`madvise(2)`](https://man7.org/linux/man-pages/man2/madvise.2.html):
> _madvise(0, 0, advice)_ will return zero iff _advice_ is supported by the kernel and can be relied on to probe for support.

Reply:
The check could be like this:
```rust
fn populate_read_is_supported() -> bool {
    use std::sync::OnceLock;
    static IS_SUPPORTED: OnceLock<bool> = OnceLock::new();
    *IS_SUPPORTED.get_or_init(|| unsafe {
        libc::madvise(std::ptr::null_mut(), 0, Advice::PopulateRead as libc::c_int) == 0
    })
}
```

Reply:
If we're running 5.14 (at least) in the cloud I don't think we should care too much about older versions.

We can, but I prefer to keep things simpler.

Reply:
In some cases we do use an older kernel version. So, lets follow your suggestion with the runtime check.

@xzfc I've tagged you in Slack where we've discussed this.

Reply:
Recap: we still have older versions somewhere.

And even if we didn't, performing this tiny check shouldn't complicate things much, since we already have the fallback code for non-Linux systems.

Reply:
I've implemented that in the last commit to this PR. It uses a new `mmap2::Advise::is_supported()` method added in https://www.github.com/RazrFalcon/memmap2-rs/pull/122.